### PR TITLE
feat(CO-58): PUT /hero/:id/action/:type endpoint

### PR DESCRIPTION
**Pull request by dyanimay**
Officially Misi's task but it turned out to be finished which started as a practice task.
thanks Réka!

### CODE REVIEW

**Review by rrtoth — DISMISSED** [review]

File: src/services/idleActionService.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+  setIdleStatus(heroId, type) {
+    if (Number.isNaN(Number(heroId))) return Promise.reject(new Error(400));
+    if (type !== ('rest' || 'scout' || 'train')) {
+      return Promise.reject(new Error(400));
+    }
+    const currentTime = new Date() / 1000;
+    let query = 'SELECT * FROM idleStatus WHERE heroId = ?;';
+    return new Promise((resolve, reject) => {
+      this.conn.query(query, [heroId, type], (err, row) => {
```
Comment (juhaszPeterDev) [inline src/services/idleActionService.js]: i dont see a second questionmark in the query, do you need the type in the input?

File: src/services/idleActionService.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        if (row.length === 0) {
+          query =
+            'INSERT INTO idleStatus(heroId, type, timestamp) VALUES(?, ?, ?);';
+          values = [heroId, type, currentTime];
+        } else if (row[0].type !== type) {
+          query =
+            'UPDATE idleStatus SET type = ?, timestamp = ? WHERE heroId = ?;';
+          values = [type, currentTime, heroId];
+        } else {
+          reject(new Error(400));
```
Comment (juhaszPeterDev) [inline src/services/idleActionService.js]: pls make this a successful return. you dont have to add additional logic, just dont return with an error.

**Review by juhaszPeterDev — CHANGES_REQUESTED** [review]

File: src/services/idleActionService.js (hunk 1 above)
Comment (dyanimay) [reply src/services/idleActionService.js]: corrected

File: src/services/idleActionService.js (hunk 2 above)
Comment (dyanimay) [reply src/services/idleActionService.js]: I changed it, but it meant to be an error for the scenario: when the user wants to select an idle action which is already running and in the database, so it was an error

**Review by juhaszPeterDev — APPROVED** [review]

**Pull request merged**